lib/application/widgets/theme/contributors_button.dart

```diff
@@ -0,0 +1,138 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+
+import 'package:memo/application/theme/theme_controller.dart';
+import 'package:memo/application/widgets/theme/link.dart';
+import 'package:memo/domain/models/contributor.dart';
+import 'package:memo/application/constants/dimensions.dart' as dimens;
+import 'package:memo/application/constants/strings.dart' as strings;
+import 'package:memo/application/utils/bottom_sheet.dart';
+
+class MultipleContributorsButton extends HookWidget {
+  const MultipleContributorsButton(this.contributors);
+
+  final List<Contributor> contributors;
+
+  void _showBottomSheet(BuildContext context, Color backgroundColor, Color imageBackgroundColor) {
+    context.showDraggableScrollableModalBottomSheet<void>(
+      child: Column(
+        children: contributors
+            .map((contributor) => Padding(
+                  padding: const EdgeInsets.symmetric(
+                    vertical: dimens.contributorsButtonVerticalPadding,
+                    horizontal: dimens.contributorsButtonHorizontalPadding,
+                  ),
+                  child: SingleContributorButton(
+                    contributor,
+                    imageRadius: dimens.contributorsLargeImageRadius,
+                    backgroundColor: backgroundColor,
+                    imageBackgroundColor: imageBackgroundColor,
+                  ),
+                ))
+            .toList(),
+      ),
+      title: strings.contributorsTitle,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final memoTheme = useTheme();
+
+    final themeColor = memoTheme.neutralSwatch.shade200;
+    final description = strings.numOfContributors(contributors.length);
+    final maxImages =
+        contributors.length > dimens.maxContributorsImages ? dimens.maxContributorsImages : contributors.length;
+
+    final contributorsImages = contributors
+        .sublist(0, maxImages)
+        .asMap()
+        .map((index, contributor) => MapEntry(
```

Comment:
Because the return value of the map method of a Map must return a Map.

![image](https://user-images.githubusercontent.com/64819163/122542163-c2b2b080-d000-11eb-99ff-39c4eed7e7b5.png)

So I guess, mapping it to a MapEntry with it's index and then getting its values it's how we achieve the final list considering an iteration over the indexes.

Maybe this will go off if I use the layoutr as suggested in other comments.

Reply:
@ggirotto

Reply:
@lucasbiancogs You can use `indexedMap` from `collection` package to do it like this:

```dart
import 'package:collection/collection.dart';

final contributorsImages = contributors.sublist(0, maxImages).mapIndexed((index, element) => _buildContributorAvatar).toList().reversed.toList()

```